url8551e9ebc1_mishika05_Salesforce_DataGen_tgz-pJ8-org_json_TestJ8/src/SalesForceClient.java

```diff
@@ -0,0 +1,185 @@
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.util.LinkedList;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.impl.conn.PoolingClientConnectionManager;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.apache.http.conn.ClientConnectionManager;
+import org.apache.http.conn.scheme.Scheme;
+import org.apache.http.conn.scheme.SchemeRegistry;
+import org.apache.http.conn.ssl.*;
+import org.apache.log4j.Logger;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.json.JSONObject;
+
+/**
+ * The SalesForceClient class generates a secure http client as well as an access token
+ * based on the credentials listed in Constants.java (i.e. environment, client_id, etc.)
+ * @author  Mishika Vora
+ */
+public class SalesForceClient {
+    public HttpClient httpclient;
+    public String accessToken;
+    public boolean isBadClient; //to make sure client is good
+    private static Logger logger = Logger.getLogger(SalesForceClient.class);
+    public MyProperties prop;
+
+
+    /**
+     * Creates a new SalesforceClient object to be passed into creating a record object
+     * Each object has an httpclient and an access token that can be access outside the class
+     */
+	public SalesForceClient(String propFile) {
+		httpclient = null;
+		accessToken = null;
+		isBadClient = false;
+		prop = new MyProperties(propFile);
+		try {
+			this.httpclient = secureHttpClient();
+			logger.info("Secure client successfully created from: " + propFile);
+		} catch (KeyManagementException e) {
+			// TODO Auto-generated catch block
+			isBadClient = true;
+			logger.error(e);
+			e.printStackTrace();
+			
+		} catch (NoSuchAlgorithmException e) {
+			// TODO Auto-generated catch block
+			isBadClient = true;
+			logger.error(e);
+			e.printStackTrace();
+		}
+		
+		try {
+			this.accessToken = retrieveAccessToken();
+			if (accessToken != "Error")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.